### PR TITLE
add field to PeerInfo struct

### PR DESCRIPTION
**Pull request by poornas**
for passing sync mode in site replication

### CODE REVIEW

File: batch-job.go [hunk 1]
```diff
@@ -205,7 +205,7 @@ type GenerateBatchJobOpts struct {
 
 // GenerateBatchJob creates a new job template from standard template
 // TODO: allow configuring yaml values
-func (adm *AdminClient) GenerateBatchJob(ctx context.Context, opts GenerateBatchJobOpts) (string, error) {
+func (adm *AdminClient) GenerateBatchJob(opts GenerateBatchJobOpts) (string, error) {
```
Comment (harshavardhana) [inline batch-job.go:208]: unrelated change please revert - pass `_ context.Context` like this, we need this for future purposes.

File: cluster-commands.go [hunk 2]
```diff
@@ -151,7 +151,18 @@ type PeerInfo struct {
 	Name     string `json:"name"`
 	// Deployment ID is useful as it is immutable - though endpoint may
 	// change.
-	DeploymentID string `json:"deploymentID"`
+	DeploymentID string     `json:"deploymentID"`
+	SyncState    SyncStatus `json:"sync"` // whether to enable| disable synchronous replication
```
Comment (harshavardhana) [inline cluster-commands.go:155]: why is this not a boolean? @poornas

File: health.go [hunk 3]
```diff
@@ -330,7 +330,7 @@ func GetOSInfo(ctx context.Context, addr string) OSInfo {
 
 // GetSysConfig returns config values from the system
 // (only those affecting minio performance)
-func GetSysConfig(ctx context.Context, addr string) SysConfig {
+func GetSysConfig(addr string) SysConfig {
```
Comment (harshavardhana) [inline health.go:333]: unrelated change please revert

File: health.go [hunk 4]
```diff
@@ -356,7 +356,7 @@ func GetSysConfig(ctx context.Context, addr string) SysConfig {
 }
 
 // GetSysServices returns info of sys services that affect minio
-func GetSysServices(ctx context.Context, addr string) SysServices {
+func GetSysServices(addr string) SysServices {
```
Comment (harshavardhana) [inline health.go:359]: unrelated change please revert

File: health.go [hunk 5]
```diff
@@ -397,7 +397,7 @@ func getSELinuxInfo() (SysService, error) {
 }
 
 // GetSysErrors returns issues in system setup/config
-func GetSysErrors(ctx context.Context, addr string) SysErrors {
+func GetSysErrors(addr string) SysErrors {
```
Comment (harshavardhana) [inline health.go:400]: unrelated change please revert

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cluster-commands.go (hunk 2 above)
Comment (harshavardhana) [reply cluster-commands.go:155]: IMO we can make `enable/disable` an `mc` level construct instead of having this via SDK. It is very simple to just have `SyncReplication bool`

Comment (poornas) [reply cluster-commands.go:155]: Passing a bool would also entail an additional flag to know if isSyncReplicationSet - otherwise an endpoint change can overwrite the sync mode setting on the server

File: health.go (hunk 5 above)
Comment (poornas) [reply health.go:400]: to address lint failures

File: cluster-commands.go (hunk 2 above)
Comment (harshavardhana) [reply cluster-commands.go:155]: Understood.

**Review by harshavardhana — APPROVED** [review]
